### PR TITLE
[improvement] better default hierarchy display

### LINKED ISSUE #11052: Calypso's hierarchy should not be reversed by default when compiling an overriden method
**Bug description**
When you select a method from a class that is different from your current class, a menu giving a reversed hierarchy of classes is provided.
I consider (and @Inao0 as well) that this is error prone, because we always have a normal hierarchy.
I am installing methods in the wrong class on a daily basis.
This is a setting with the (arguably) wrong default behavior.

**To Reproduce**
1. Go to a class with methods in the super class
2. Click on the checkbox to have methods inherited by the current class
3. Select a method
4. Change something
5. Try to save
6. A dialog appears showing you the reversed hierarchy

**Expected behavior**
The hierarchy should be in the regular order

**Screenshots**
Current default
Screenshot 2022-03-25 at 11 19 20

Expected
Screenshot 2022-03-25 at 11 18 59

**Version information:**
Pharo 10

**Expected development cost**
Trivial, it's a default setting to update.

**Additional context**
The class to preselect in the list is another issue, that we will open later.

### PR DESCRIPTION
**Pull request by hogoww**
Fixes #11052 
One of those settings we don't know about ! :D

### CODE REVIEW

**Comment by MarcusDenker** [comment]
Failing test:

```
Error
Given Collections do not match!  additions : an Array(a StoredSetting('#Calypso#isDefaultHierarchyForMethodVisibility'))  missing: #()
Stacktrace
TestFailure
Given Collections do not match!
	additions : an Array(a StoredSetting('#Calypso#isDefaultHierarchyForMethodVisibility'))
	missing: #()
SystemSettingsPersistenceTest(TestAsserter)>>assert:description:resumable:
SystemSettingsPersistenceTest(TestAsserter)>>assert:description:
SystemSettingsPersistenceTest(TestAsserter)>>assertCollection:hasSameElements:
SystemSettingsPersistenceTest>>testDefaultImageDoesNotStoreAnySetting
SystemSettingsPersistenceTest(TestCase)>>performTest
```

**Comment by hogoww** [comment]
After investigation, it's because the setting in the setting browser is different from the one that is defined in the method.
I am unsure of where the setting browser is initialised however

**Comment by hogoww** [comment]
After investigation with @demarey there seems to be an old config file that is taken into account although it should.
We should probably add the --no-preference flag to the Travis file, I was asked to ping @tesonep about it.

**Comment by hogoww** [comment]
As asked by @estebanlm, changed target to p11

**Comment by hogoww** [comment]
same failing test.
see previous comment

**Comment by hogoww** [comment]
We fixed the test case that was failing with @demarey on this sprint to not use the current system but rather a mock.
We should now be able to change default parameters more easily.

**Review by Bajger — APPROVED** [review]
Where is pre-selection of currently used class? You want to have it in separate PR?

**Comment by hogoww** [comment]
> Where is pre-selection of currently used class? You want to have it in separate PR?

I was planning to investigate this on the sprint following the integration of this PR.
But it's been a while.
With Iona, we send a mail about this to see if people had a preference about this: https://lists.pharo.org/empathy/thread/HTIE7ZPSIUTBWL5T5SBAVTWZKCU2VN7H

<blockquote><div><strong><a href="https://lists.pharo.org/empathy/thread/HTIE7ZPSIUTBWL5T5SBAVTWZKCU2VN7H">Empathy List Archives</a></strong></div><div>Empathy mailing list archives</div></blockquote>

**Review by jordanmontt — APPROVED** [review]

**Comment by hogoww** [comment]
I do not intent to pursue this any further.
@MarcusDenker Do you mind reviewing it?
It's really just a setting change, and a test update, there's nothing to it...
Plus it was already approved by Sebastian.

In the case of big hierarchy, it doesn't seem to matter because it only shows you 2 classes anyway.
But at least, they'll be in order, and that tool will finally be usable without me thinking for a minute each time I have to use it.

**Comment by MarcusDenker** [comment]
I am trying my best to review PRs, but there are just too many for me to be able to promise anything...

**Comment by hogoww** [comment]
> I am trying my best to review PRs, but there are just too many for me to be able to promise anything...

I know, I'm asking specifically for this one because it's super short.
It's literally just a default setting change.
Couldn't be shorter.

On top of that we had to fix a test case, which I agree is harder to review.
But that was Christophe's doing, and he works well =)

File: src/System-Settings-Tests/SystemSettingsPersistenceTest.class.st [hunk 1]
```diff
@@ -103,11 +103,12 @@ SystemSettingsPersistenceTest >> testDefaultImageDoesNotStoreAnySetting [
 	| nodesToStore |
 	"An image with all default values should not save any setting.
 	The settings only should be saved if changed from default value"
-
+"MockSettings foo."
```
Comment (guillep) [inline src/System-Settings-Tests/SystemSettingsPersistenceTest.class.st]: Leftover?

**Review by guillep — CHANGES_REQUESTED** [review]
Could we remove the leftover comment?
Otherwise, the change is good to go.

**Comment by jordanmontt** [comment]
Now there are merge conflicts

File: src/System-Settings-Tests/SystemSettingsPersistenceTest.class.st (hunk 1 above)
Comment (hogoww) [reply src/System-Settings-Tests/SystemSettingsPersistenceTest.class.st]: I guess, it's been too long.
Thanks for spotting it.

**Comment by hogoww** [comment]
> Now there are merge conflicts

Solved. 
It was conflicting because of Astares' recent cleanup wave.

**Review by guillep — APPROVED** [review]
Good!

**Pull request merged**